File: WorkManagerJava/app/src/main/java/com/websarva/wings/android/workmanagerjava/FinishCountWithIdWorker.java
package com.websarva.wings.android.workmanagerjava;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Data;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class FinishCountWithIdWorker extends Worker {
	public FinishCountWithIdWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
		super(context, workerParams);
	}

	@NonNull
	@Override
	public Result doWork() {
		Data inputData = getInputData();
		String[] idList = inputData.getStringArray("id");
		StringBuffer sb = new StringBuffer();
		sb.append("CountWithIdWorkerが終了しました。");
		sb.append("\n終了したid: ");
		for(String id: idList) {
			sb.append("\n");
			sb.append(id);
		}
		Log.i("FinishCountWithIdWorker", sb.toString());
		return Result.success();
	}
}

File: WorkManagerKotlin/app/src/main/java/com/websarva/wings/android/workmanagerkotlin/FinishCountWithIdWorker.kt
package com.websarva.wings.android.workmanagerkotlin

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters

class FinishCountWithIdWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
	override fun doWork(): Result {
		val idList = inputData.getStringArray("id")
		val sb = StringBuffer()
		if (idList != null) {
			for(id in idList) {
				sb.append("\n${id}")
			}
		}
		Log.i("FinishCountWithIdWorker","CountWithIdWorkerが終了しました。\n終了したid: ${sb}")
		return Result.success()
	}
}
